### PR TITLE
Accept room names for robot.messageRoom

### PR DESCRIPTION
**Pull request by mdelagrange**
* [X] I've read and understood the [Contributing guidelines](./CONTRIBUTING.md) and have done my best effort to follow them.
* [X] I've read and agree to the [Code of Conduct](./CODE_OF_CONDUCT.md).
* [X] I've been mindful about doing atomic commits, adding documentation to my changes, not refactoring too much.
* [X] I've a descriptive title and added any useful information for the reviewer. Where appropriate, I've attached a screenshot and/or screencast (gif preferrably).
* [X] I've written tests to cover the new code and functionality included in this PR.
* [X] I've read, agree to, and signed the [Contributor License Agreement (CLA)](https://docs.google.com/a/slack-corp.com/forms/d/1q_w8rlJG_x_xJOoSUMNl7R35rkpA7N6pUkKhfHHMD9c/viewform).

#### PR Summary
Previous to hubot-slack 4.0.0, `robot.messageRoom` would accept room names.  This PR translates room names to room ids, in order to improve compatibility with existing hubot scripts.

#### Test strategy
Write a hubot script that sends a message to a room by name, e.g.:
```coffeescript
module.exports = (robot) ->

  robot.respond /test messageRoom/, (msg) ->
    room = 'general'
    robot.messageRoom room, 'hi there!'
```
This will not work with hubot-slack 4.0.2.

### CODE REVIEW

**Comment by Mezzle** [comment]
:+1:

Fixes: #349

**Comment by Mezzle** [comment]
Actually, personally I'd prefer this with the `#` channel prefix demarking the fact that it's a channel name, or using the `RECOGNIZED_API_TYPE_PREFIXES` to determine it's not an API prefix?

File: src/client.coffee [hunk 1]
```diff
@@ -72,13 +72,19 @@ class SlackClient
   ###
   send: (envelope, message) ->
     message = @format.outgoing(message)
+    room = envelope.room
+    if !(room.match /C0[A-Z0-9]{7}/)
```
Comment (Mezzle) [inline src/client.coffee]: Channels begin with `C` not `C0`

see https://github.com/slackhq/node-slack-sdk/blob/master/lib/models/helpers.js#L28

**Comment by mdelagrange** [comment]
> Actually, personally I'd prefer this with the # channel prefix demarking the fact that it's a channel name, or using the RECOGNIZED_API_TYPE_PREFIXES to determine it's not an API prefix?

Actually `getChannelByName(room)` works with or without the hash, so your example should work fine.

**Comment by mdelagrange** [comment]
> Channels begin with C not C0

Since Slack channels never have capital letters, I'm just testing for those now.  That should be a reliable test.

**Comment by Mezzle** [comment]
@mdelagrange I would expect that if I put in 'GENERAL'  that it would push to `#general` ?

**Comment by mdelagrange** [comment]
That wasn't the behavior in hubot-slack < 4.0.0.  Channel names were case sensitive in previous versions.  I'd like to avoid an unnecessary API call, so that's my reason for sticking with case sensitivity.

**Comment by DEGoodmanWilson** [comment]
LGTM

**Comment by markmcintyre** [comment]
@DEGoodmanWilson: I just tested out this fix, which works great; but (and forgive me if I'm misunderstanding the process here) the package.json version number hasn't been updated. This means the latest version on NPM doesn't have this fix included. Is that intentional?

**Comment by DEGoodmanWilson** [comment]
Right, I don't think I released this. Last two weeks have been a bit hectic :( Will look into this ASAP

**Pull request merged**